main.cpp

```diff
@@ -110,16 +110,18 @@ struct Numeric
  public:
 
     Numeric(NumType nuNum) : value( std::make_unique<Type>(nuNum) ) { }
-    ~Numeric(){ }
+    
 
+    ~Numeric() = default;
     //rule of 5 funcs
-    Numeric(Type&& num){ *value = std::move(num); }
-    Numeric& operator=(NumType&& num)
+    Numeric(Numeric&& num){ *value = std::move(num); }
```

Comment:
- use the member initializer list for constructors, not the in-constructor variable assignment.
- don't dereference.  you're forcing `num` to be converted to `Type`. 
```
Numeric(Numeric&& num) : value(std::move(num.value)) { } 
```
- [ ] revise